src/main/java/at/hannibal2/skyhanni/features/commands/OpenLastStorage.kt

```diff
@@ -0,0 +1,125 @@
+package at.hannibal2.skyhanni.features.commands
+
+import at.hannibal2.skyhanni.events.MessageSendToServerEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.ChatUtils.senderIsSkyhanni
+import at.hannibal2.skyhanni.utils.HypixelCommands
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+@SkyHanniModule
+object OpenLastStorage {
+    private enum class StorageType {
+        ENDER_CHEST, BACKPACK
+    }
+
+    private var lastBackpack = -1
+    private var lastEnderChest = -1
+    // Default to Ender Chest as last storage type, since every profile on any
+    // account has at least one partial ender chest page unlocked
+    private var lastStorageType = StorageType.ENDER_CHEST
+
+    private fun openLastBackPack() {
+        HypixelCommands.openBackpack(lastBackpack)
+        ChatUtils.chat("Opened last backpack $lastBackpack.")
+    }
+
+    private fun openLastEnderChest() {
+        HypixelCommands.openEnderChest(lastEnderChest)
+        ChatUtils.chat("Opened last ender chest $lastEnderChest.")
+    }
+
+    @SubscribeEvent
+    fun onMessageSendToServer(event: MessageSendToServerEvent) {
+        if (event.senderIsSkyhanni()) return
+
+        if (handleBackpack(event)) return
+        if (handleEnderChest(event)) return
+
+        // Non-Hypixel combined "last opened" command for backpack + ender chest
+        if (!event.message.startsWith("/lastopened", ignoreCase = true) &&
+            // cannot use startsWith because of /locraw
+            !event.message.equals("/lo", ignoreCase = true)
+        ) {
+            return
+        }
+
+        event.isCanceled = true
+
+        if (lastStorageType == StorageType.ENDER_CHEST) {
+            openLastEnderChest()
+        } else if (lastStorageType == StorageType.BACKPACK) {
+            openLastBackPack()
+        }
+    }
+
+    private fun handleBackpack(event: MessageSendToServerEvent): Boolean {
+        if (!event.message.startsWith("/backpack", ignoreCase = true) &&
+            !event.message.startsWith("/bp", ignoreCase = true)
+        ) {
+            return false
+        }
+
+        if (event.message.startsWith("/backpack -", ignoreCase = true) ||
```

Comment:
instead of ignore case everywhere, you can make the message to lower case once at the beginning

Reply:
Done